docs/website/docs/community/blog/posts/data-tiling-with-encoding-specialization.md

```diff
@@ -0,0 +1,590 @@
+
+The pass set the encodings on matmul operands, and propagate the encodings through the fill op. Because we want to fuse the fill op into the matmul kernel. The `round_dims_to` is redundant in the configuration, you can ignore it for now.
+
+```mlir
+// -----// IR Dump After SetEncodingPass (iree-dispatch-creation-set-encoding) //----- //
+
+#map = affine_map<(d0, d1, d2) -> (d0, d2)>
+#map1 = affine_map<(d0, d1, d2) -> (d2, d1)>
+#map2 = affine_map<(d0, d1, d2) -> (d0, d1)>
+#encoding = #iree_encoding.encoding<operand_index = 0 : index, op_type =  matmul, element_types = [f32, f32, f32], user_indexing_maps = [#map, #map1, #map2], round_dims_to = array<i64: 32, 32, 32>>
+#encoding1 = #iree_encoding.encoding<operand_index = 1 : index, op_type =  matmul, element_types = [f32, f32, f32], user_indexing_maps = [#map, #map1, #map2], round_dims_to = array<i64: 32, 32, 32>>
+#encoding2 = #iree_encoding.encoding<operand_index = 2 : index, op_type =  matmul, element_types = [f32, f32, f32], user_indexing_maps = [#map, #map1, #map2], round_dims_to = array<i64: 32, 32, 32>>
+module {
+  util.func public @foo(%arg0: !hal.buffer_view, %arg1: !hal.buffer_view) -> !hal.buffer_view attributes {iree.abi.stub, iree.reflection = {iree.abi.declaration = "sync func @foo(%input0: tensor<?x?xf32>, %input1: tensor<?x?xf32>) -> (%output0: tensor<?x?xf32>)"}} {
+    %cst = arith.constant 0.000000e+00 : f32
+    %0 = hal.buffer_view.dim<%arg0 : !hal.buffer_view>[0] : index
+    %1 = hal.buffer_view.dim<%arg0 : !hal.buffer_view>[1] : index
+    %2 = hal.tensor.import %arg0 "input0" : !hal.buffer_view -> tensor<?x?xf32>{%0, %1}
+    %3 = hal.buffer_view.dim<%arg1 : !hal.buffer_view>[0] : index
+    %4 = hal.buffer_view.dim<%arg1 : !hal.buffer_view>[1] : index
+    %5 = hal.tensor.import %arg1 "input1" : !hal.buffer_view -> tensor<?x?xf32>{%3, %4}
+    %6 = iree_encoding.set_encoding %2 : tensor<?x?xf32> -> tensor<?x?xf32, #encoding>
+    %7 = iree_encoding.set_encoding %5 : tensor<?x?xf32> -> tensor<?x?xf32, #encoding1>
+    %8 = tensor.empty(%0, %4) : tensor<?x?xf32, #encoding2>
+    %9 = linalg.fill ins(%cst : f32) outs(%8 : tensor<?x?xf32, #encoding2>) -> tensor<?x?xf32, #encoding2>
+    %10 = linalg.matmul ins(%6, %7 : tensor<?x?xf32, #encoding>, tensor<?x?xf32, #encoding1>) outs(%9 : tensor<?x?xf32, #encoding2>) -> tensor<?x?xf32, #encoding2>
+    %11 = iree_encoding.unset_encoding %10 : tensor<?x?xf32, #encoding2> -> tensor<?x?xf32>{%0, %4}
+    %12 = hal.tensor.export %11 "output0" : tensor<?x?xf32>{%0, %4} -> !hal.buffer_view
+    util.return %12 : !hal.buffer_view
+  }
+}
+```
+
+### Outline dispatches
+
+There are four dispatches in tatal. `dispatch_0` packs the LHS operand, `dispatch_1` packs the RHS operand, `dispatch_2` is `fill + gemm`, and `dispatch_3` unpacks the result of matmul.
+
+#### IR dump after the pass
+
+```mlir
+// -----// IR Dump After OutlineDispatchRegionsPass (iree-flow-outline-dispatch-regions) //----- //
```

Comment:
These are pretty large code blocks. If you want to highlight something in particular, you could add line numbers and/or line highlighting

For example, adding
````
```mlir linenums="1" hl_lines="18-27"
````
here produces this code block: 
![image](https://github.com/user-attachments/assets/2ae86f6a-e21c-4a84-a60e-f89bbd660303)

(Yes, syntax highlighting would be nice too 🙃)

Docs:

* https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-line-numbers
* https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#highlighting-specific-lines